server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/FileOperationProcess.java

```diff
@@ -102,17 +106,18 @@ private void checkFileOpenedBeforeOperation(
         messageTemplate);
   }
 
-  private boolean fileIsExternal(FileOperationStatementNode node) {
-    if (!node.getNearestParentByType(PROGRAM).isPresent()) return false;
-    Optional<FileDescriptionNode> fileDescriptionNode =
-        node.getNearestParentByType(PROGRAM)
-            .get()
-            .getDepthFirstStream()
-            .filter(FileDescriptionNode.class::isInstance)
-            .map(FileDescriptionNode.class::cast)
-            .filter(fdNode -> fdNode.getName().equalsIgnoreCase(node.getFilename().getName()))
-            .findFirst();
-    return fileDescriptionNode.map(FileDescriptionNode::isExternal).orElse(false);
+  private boolean fileIsExternal(FileOperationStatementNode node, ProgramNode programNode) {
+    Node fd = programNode.getDepthFirstFirstNode(n -> {
+      if (n.getClass() != FileDescriptionNode.class) {
+        return false;
+      }
+      FileDescriptionNode fdNode = (FileDescriptionNode) n;
+      if (!fdNode.isExternal()) {
+          return false;
+      }
+      return fdNode.getName().equalsIgnoreCase(node.getFilename().getName());
```

Comment:
Not sure if this is a meaningful difference, but the original code found the node by name and the examined the `isExternal` property, the new one tries to find the external node no matter what.

Reply:
I think it should be equivalent to original implementation:
we need to make sure that there us no file description node with file name and isExternal flag on. So we don't care about not external nodes (quick check) and for external we search for the file name in them.